packages/dialect-react/components/DialectContext/index.tsx

```diff
@@ -274,10 +274,13 @@ export const DialectProvider = (props: PropsType): JSX.Element => {
     const hasNewMessage =
       wallet &&
       dialect?.dialect &&
-      messages.length !== dialect.dialect.messages.length;
+      (messages.length !== dialect.dialect.messages.length 
+        // Could be there performacne issue to calc hash for long array?
+        || getMessageHash(dialect.dialect.messages) !== getMessageHash(messages));
```

Comment:
If it's a heavy operation to hash (even though I think that sha1 shouldn't be too heavy, we don't have arrays of astronomical size), feel free to useMemo it and use that.

Also, I think that we don't need to the check the length any more since hashing will prove the change. wdyt?

Reply:
Yes, message.length could be removed in this case. Will updated PR

Reply:
Maybe a silly question, but why we introduced the condition here to update messages only if lenght is not matched, the fact `setMessages` is called in useEffect with `wallet, dialect?.dialect` dependencies should already do the trick, no?

@fsher